pkg/fanal/analyzer/language/julia/pkg/pkg.go

```diff
@@ -0,0 +1,189 @@
+			log.Logger.Warnf("Unable to parse %q to analyze dependencies: %s", filepath.Join(filepath.Dir(path), types.JuliaProject), err)
+		}
+
+		sort.Sort(app.Libraries)
+		apps = append(apps, *app)
+		return nil
+	})
+	if err != nil {
+		return nil, xerrors.Errorf("julia walk error: %w", err)
+	}
+
+	return &analyzer.AnalysisResult{
+		Applications: apps,
+	}, nil
+}
+
+func (a juliaAnalyzer) Required(filePath string, _ os.FileInfo) bool {
+	fileName := filepath.Base(filePath)
+	return slices.Contains(requiredFiles, fileName)
+}
+
+func (a juliaAnalyzer) Type() analyzer.Type {
+	return analyzer.TypeJulia
+}
+
+func (a juliaAnalyzer) Version() int {
+	return version
+}
+
+func (a juliaAnalyzer) parseJuliaManifest(path string, r io.Reader) (*types.Application, error) {
+	return language.Parse(types.Julia, path, r, a.lockParser)
+}
+
+func analyzeDependencies(fsys fs.FS, dir string, app *types.Application) error {
+	deps, devDeps, err := parseDependencies(fsys, dir)
+	if err != nil {
+		return err
+	}
+
+	pkgs := walkDependencies(deps, app.Libraries, false)
+	devPkgs := walkDependencies(devDeps, app.Libraries, true)
+	pkgs = lo.Assign(devPkgs, pkgs)
+	app.Libraries = maps.Values(pkgs)
+	return nil
+}
+
+func parseDependencies(fsys fs.FS, dir string) (map[string]string, map[string]string, error) {
+	projectPath := filepath.Join(dir, types.JuliaProject)
+	project, err := parseJuliaProject(fsys, projectPath)
+	if errors.Is(err, fs.ErrNotExist) {
+		log.Logger.Debugf("Julia: %s not found", projectPath)
+		return nil, nil, nil
+	} else if err != nil {
+		return nil, nil, xerrors.Errorf("unable to parse %s: %w", projectPath, err)
+	}
+	return project.Dependencies, project.Extras, nil
+}
+
+// Parses Project.toml
+func parseJuliaProject(fsys fs.FS, path string) (Project, error) {
+	proj := Project{}
+	f, err := fsys.Open(path)
+	if err != nil {
+		return proj, xerrors.Errorf("file open error: %w", err)
+	}
+	defer func() { _ = f.Close() }()
+
+	if _, err = toml.NewDecoder(f).Decode(&proj); err != nil {
+		return proj, xerrors.Errorf("decode error: %w", err)
+	}
+	return proj, nil
+}
+
+// Marks the given direct dependencies as direct, then marks those packages' dependencies as indirect.
+// Marks all encountered packages' Dev flag according to `dev`.
+// Modifies the packages in `allPackages`.
+func walkDependencies(directDeps map[string]string, allPackages types.Packages, dev bool) map[string]types.Package {
+	pkgsByID := lo.SliceToMap(allPackages, func(pkg types.Package) (string, types.Package) {
+		return pkg.ID, pkg
+	})
+
+	// Identify direct dependencies
+	// Everything in `directDeps` is assumed to be direct
+	visited := map[string]types.Package{}
+	for _, uuid := range directDeps {
+		if pkg, ok := pkgsByID[uuid]; ok {
+			pkg.Indirect = false
+			pkg.Dev = dev
+			visited[pkg.ID] = pkg
+		}
+	}
+
+	// Identify indirect dependencies
+	for _, pkg := range visited {
+		walkIndirectDependencies(pkg, pkgsByID, visited, dev)
+	}
+
+	return visited
```

Comment:
I think we don't need to return map here.
```suggestion
	return maps.Values(visited)
```